File: frontend/src/types/enforcementRoi.ts
/** 维权ROI计算器类型定义 */

export interface DecisionTreeResult {
  recommended_actions: Array<{
    action_key: string
    name_zh: string
    estimated_cost: [number, number]
    expected_duration_days: [number, number]
    win_rate: number
    note_zh?: string
  }>
  primary_recommendation: dict
  reasoning: string
}

export interface RoiPrediction {
  expected_cost: number
  expected_duration_days: number
  win_probability: number
  expected_compensation: number
  net_return: number
  roi_percent: number
  risk_level: 'low' | 'medium' | 'high'
}

export interface DefenseTier {
  id: string
  tier_key: string
  tier_name_zh: string
  monthly_cost_low?: number
  monthly_cost_high?: number
  annual_cost_low?: number
  annual_cost_high?: number
  features?: string[]
  description_zh?: string
  recommended_for?: string
}

export interface CaseReference {
  id: string
  infringement_type: string
  target_platform: string
  typical_cost_range_low?: number
  typical_cost_range_high?: number
  resolution_time_days_low?: number
  resolution_time_days_high?: number
  win_rate_percent?: number
  avg_compensation_yuan?: number
  roi_tier?: string
  description_zh?: string
}

export interface UserCasesSummary {
  cases: Array<{
    id: string
    infringement_type: string
    target_platform: string
    estimated_loss_yuan: number
    cost_yuan: number
    compensation_received_yuan: number
    outcome: string
    roi_percent?: number
    created_at: string
  }>
  summary: {
    total_cases: number
    successful_cases: number
    success_rate_percent: number
    total_cost: number
    total_compensation: number
    overall_roi_percent?: number
  }
}

File: backend/app/schemas/enforcement_roi.py
"""维权ROI计算器 Pydantic schemas."""

from pydantic import BaseModel
from typing import Optional
from datetime import date


class EnforcementCaseCreate(BaseModel):
    """创建维权案例请求."""
    work_id: Optional[str] = None
    infringement_type: str
    target_platform: str
    estimated_loss_yuan: float
    action_taken: str
    cost_yuan: float = 0
    compensation_received_yuan: float = 0
    outcome: str
    notes: Optional[str] = None


class EnforcementCaseResponse(BaseModel):
    """维权案例响应."""
    id: str
    user_id: str
    work_id: Optional[str] = None
    infringement_type: str
    target_platform: str
    estimated_loss_yuan: float
    cost_yuan: float
    time_to_resolve_days: Optional[int] = None
    compensation_received_yuan: float
    outcome: str
    roi_percent: Optional[float]
    created_at: str


class CaseReferenceSchema(BaseModel):
    """参考案例响应."""
    id: str
    infringement_type: str
    target_platform: str
    typical_cost_range_low: Optional[float] = None
    typical_cost_range_high: Optional[float] = None
    resolution_time_days_low: Optional[int] = None
    resolution_time_days_high: Optional[int] = None
    win_rate_percent: Optional[float] = None
    avg_compensation_yuan: Optional[float] = None
    roi_tier: Optional[str] = None


class DecisionTreeResult(BaseModel):
    """决策树推荐结果."""
    recommended_actions: list[dict]
    primary_recommendation: dict
    reasoning: str


class RoiPredictorRequest(BaseModel):
    """ROI预测器请求."""
    work_value_yuan: float
    infringement_type: str
    target_platform: str
    action_type: str  # "cease_desist" / "platform_complaint" / "civil_lawsuit" / "criminal_report"


class RoiPrediction(BaseModel):
    """ROI预测结果."""
    expected_cost: float
    expected_duration_days: int
    win_probability: float
    expected_compensation: float
    net_return: float
    roi_percent: float
    risk_level: str  # "low" / "medium" / "high"


class DefenseTierSchema(BaseModel):
    """防御预算层级响应."""
    id: str
    tier_key: str
    tier_name_zh: str
    monthly_cost_low: Optional[float] = None
    monthly_cost_high: Optional[float] = None
    annual_cost_low: Optional[float] = None
    annual_cost_high: Optional[float] = None
    features: Optional[list[str]] = None
    description_zh: Optional[str] = None
